src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const formatString = sourceString.replace(/;\s*;/g, ' ').trim();
+
+  const stringToArray = formatString.split(';');
+
+  const result = {};
+
+  stringToArray.map((element) => {
+    const [key, value] = element.split(':', 2);
+
+    if (value !== undefined) {
+      result[key.trim()] = value.trim();
+    }
+  });
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. Here, `map` is used incorrectly because it returns a new array, but you are using it to modify `result`. Instead, you should use `forEach` or other methods that are intended for side effects.